### ISSUE TITLE
[Blocker] Missing dependency on Maven central

### ISSUE THREAD

**Issue by claytonneal** [galenframework/galen#208]
Hiya,

Galen is dependant on Rainbow4j which is dependant on a library called jai-core v1.1.3
This librarys jar file is not present in maven central
This gives an problem resolving all dependencies when using Galen

Thanks,

**Comment by ishubin** [comment]
Which Java version are you using? Because for me it works fine with version 1.7 and 1.6

**Comment by claytonneal** [comment]
Take a look at this:
http://search.maven.org/#artifactdetails|javax.media|jai_core|1.1.3|jar
You will see that maven central has no JAR for this dependency of Rainbow4JAnd as Galen depends on Rainbow4J, if you delete your local maven repository and then do a maven install, it will fail
MY solution was to host the all-in-one Galen jar in our nexus repository
Thanks.

**Comment by ishubin** [comment]
hm... that is not good. I am will try to find a way to fix. The thing is I couldn't use the standard ImageIO lib because it was giving incorrect color scheme in some cases which makes it useless for Galen. The JAI worked fine. But if it is impossible to get it from maven central then it is a problem

**Comment by claytonneal** [comment]
The full jar download contains it - so for me i just put the full jar into our nexus repository. But it means anyone just referencing the pom will get a problem...

**Comment by ishubin** [comment]
I just checked the mave repo and here is what I see. rainbow4j has a dependency on ant-jai. And ant-jai has a dependency on jai-core and in the pom.xml it has declared the working repo where the jai-core is located (http://search.maven.org/#artifactdetails%7Corg.apache.ant%7Cant-jai%7C1.9.3%7Cjar) . Could it be that in you particular case your maven doesn't use that repository for downloading the jai-core?
Because there are now a lot of people who use Galen as a maven dependency in their Java tests and they never complained about this

**Comment by claytonneal** [comment]
Yeah your dependant on the ant jar which is dependant on the missing one

**Comment by hypery2k** [comment]
within the pom for ant-jai you'll find this:

```
!-- Central has javax.media:jai-core:1.1.3 but only com.sun.media:jai-codec:1.1.2_01 --><repositories><repository><id>jboss</id><name>JBoss</name><url>https://repository.jboss.org/nexus/content/groups/public/</url></repository></repositories>
```

Seems some issues with resolving this repository for you.
But maybe we should check if we can solve this to more clean dependency, see https://github.com/galenframework/rainbow4j/issues/1

**Comment by ishubin** [comment]
@hypery2k I checked that repository and I see that it contains the jai-core.jar file in it https://repository.jboss.org/nexus/content/groups/public/javax/media/jai-core/1.1.3/

There are definitely some issues with resolving this repository. Could be related to a specific maven version. On the other hand I still think it is a problem. I will take a look if I can put this jar into maven central

**Comment by hypery2k** [comment]
i've also checked if it's possible to add the jar to central. But the license is not allowing that. I currently working on bundling the jar directly as local jar into rainbow4j

**Comment by ishubin** [comment]
@hypery2k Ok, I think that could be an issue. Perhaps we should take a look for some other library for imaging

**Comment by ishubin** [comment]
I have found out that actually we don't need JAI and can just rely on ImageIO. It seems that used the wrong color conversion before. I will run some image tests later today and if they all work I will release the version without JAI

**Comment by claytonneal** [comment]
Many thanks

**Comment by ishubin** [comment]
@claytonneal I have fixed this in version 1.5.3. Just pushed to sonatype repository. Basically I changed the code to work with ImageIO and got rid of JAI dependency
